test/production/minimal-mode-response-cache/index.test.ts

```diff
@@ -0,0 +1,122 @@
+import glob from 'glob'
+import fs from 'fs-extra'
+import { join } from 'path'
+import cheerio from 'cheerio'
+import { createNext, FileRef } from 'e2e-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+import {
+  killApp,
+  findPort,
+  renderViaHTTP,
+  initNextServerScript,
+} from 'next-test-utils'
+
+describe('minimal-mode-response-cache', () => {
+  let next: NextInstance
+  let server
+  let appPort
+
+  beforeAll(async () => {
+    // test build against environment with next support
+    process.env.NOW_BUILDER = '1'
+
+    next = await createNext({
+      files: new FileRef(join(__dirname, 'app')),
+    })
+    await next.stop()
+
+    await fs.move(
+      join(next.testDir, '.next/standalone'),
+      join(next.testDir, 'standalone')
+    )
+    for (const file of await fs.readdir(next.testDir)) {
+      if (file !== 'standalone') {
+        await fs.remove(join(next.testDir, file))
+        console.log('removed', file)
+      }
+    }
+    const files = glob.sync('**/*', {
+      cwd: join(next.testDir, 'standalone/.next/server/pages'),
+      dot: true,
+    })
+
+    for (const file of files) {
+      if (file.endsWith('.json') || file.endsWith('.html')) {
+        await fs.remove(join(next.testDir, '.next/server', file))
+      }
+    }
+
+    const testServer = join(next.testDir, 'standalone/server.js')
+    await fs.writeFile(
+      testServer,
+      (await fs.readFile(testServer, 'utf8'))
+        .replace('console.error(err)', `console.error('top-level', err)`)
+        .replace('conf:', 'minimalMode: true,conf:')
+    )
+    appPort = await findPort()
+    server = await initNextServerScript(
+      testServer,
+      /Listening on/,
+      {
+        ...process.env,
+        PORT: appPort,
+      },
+      undefined,
+      {
+        cwd: next.testDir,
+      }
+    )
+  })
+  afterAll(async () => {
+    await next.destroy()
+    if (server) await killApp(server)
+  })
+
+  it('should have correct responses', async () => {
+    const html = await renderViaHTTP(appPort, '/')
+    expect(html.length).toBeTruthy()
+
+    for (const { path, matchedPath, query, asPath, pathname } of [
```

Comment:
Nit pick: You can use jest [`.each()`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) which will print much easier to read error message when the assertions fail.

Example usage: https://github.com/vercel/next.js/blob/b04cdc9f2f21912d1235c83544318d14bcb9739f/test/development/middleware-warnings/index.test.ts#L23